### PR TITLE
feat: TM-452 change new dates format DD/MM/YYYY

### PR DESCRIPTION
**Pull request by pachonjcl**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [X] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [X] Check the commit's or even all commits' message styles matches our requested structure.
- [X] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: [TM-452](https://gfw.atlassian.net/browse/TM-452)

## What is the new behavior?

- Some places in the application were using different date format, this PR should address those mismatches

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/admin/modules/organisations/components/OrganisationsList.tsx [hunk 1]
```diff
@@ -78,7 +78,8 @@ const ApplicationDataGrid = () => {
       </Tabs>
       <Divider />
       <Datagrid>
-        <DateField source="created_at" label="Date Created" />
+        {/* format change Organisation TM-452 */}
```
Comment (pachonjcl) [inline src/admin/modules/organisations/components/OrganisationsList.tsx:81]: Remove this comment please

File: src/admin/modules/application/components/ApplicationTabs.tsx [hunk 2]
```diff
@@ -17,6 +17,29 @@ import { getCustomFormSteps, normalizedFormDefaultValue } from "@/helpers/custom
...
+  const formatDateString = (inputDateString: string) => {
+    try {
+      const dateObject = new Date(inputDateString);
+      const formattedDay = dateObject.getUTCDate().toString().padStart(2, "0");
+      const formattedMonth = (dateObject.getUTCMonth() + 1).toString().padStart(2, "0");
+      const formattedYear = dateObject.getUTCFullYear();
+      const formattedDate = `${formattedDay}/${formattedMonth}/${formattedYear}`;
+      return formattedDate;
+    } catch (error) {
+      // Manejar el error, por ejemplo, puedes retornar "Invalid Date"
```
Comment (pachonjcl) [inline src/admin/modules/application/components/ApplicationTabs.tsx:30]: Do not leave Spanish comments

**Review by pachonjcl — COMMENTED** [review]
Please, review the `formatDateString` and make sure that there is no other way of handling that. If there is no other function that makes the same thing? please keep this function in other file and use a better naming to not get confused with other.

File: src/hooks/useVerifyDate.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  function isDateStringValid(dateString: string): boolean {
+    if (!isNaN(Number(dateString))) {
+      return false;
+    }
+    const timestamp = Date.parse(dateString);
+    return !isNaN(timestamp);
+  }
+
+  return { formatDateLocalString, formatDateString, isDateStringValid };
```
Comment (roguenet) [inline src/hooks/useVerifyDate.ts:40]: The code looks good to me aside from one thing: have we considered bringing in a library to handle date parsing and display? Luxon for instance is very powerful, and date parsing/display is fraught enough that it's really easy to miss edge cases and end up with a buggy front end.

Comment (roguenet) [reply src/hooks/useVerifyDate.ts:40]: Oh, and as I read this a bit closer; I don't see any reason for this to be a hook. It's not providing any contextual information for a component, and would be better done as a simply utility file with these three functions exported (unless we add a date math library like Luxon, in which case this is likely superfluous)

File: src/admin/modules/application/components/ApplicationTabs.tsx [hunk 4]
```diff
@@ -8,15 +8,18 @@ import {
...
 } from "react-admin";
-import { Else, If, Then, When } from "react-if";
+import { When } from "react-if";
 
+import { convertDateformat } from "@/admin/apiProvider/utils/dateFormat";
 import List from "@/components/extensive/List/List";
 import { FormSummaryRowProps, useGetFormEntries } from "@/components/extensive/WizardForm/FormSummaryRow";
 import { ApplicationRead, FormSubmissionRead } from "@/generated/apiSchemas";
 import { getCustomFormSteps, normalizedFormDefaultValue } from "@/helpers/customForms";
+// import { useVerifyDate } from "@/hooks/useVerifyDate";
```
Comment (pachonjcl) [inline src/admin/modules/application/components/ApplicationTabs.tsx:18]: remove comment

File: src/admin/modules/application/components/ApplicationTabs.tsx [hunk 5]
```diff
@@ -8,15 +8,18 @@ import {
...
 
 const ApplicationTabRow = ({ index, ...props }: FormSummaryRowProps) => {
   const entries = useGetFormEntries(props);
+
```
Comment (pachonjcl) [inline src/admin/modules/application/components/ApplicationTabs.tsx:22]: do not add innecesary blank lines

File: src/components/extensive/WizardForm/FormSummaryRow.tsx [hunk 6]
```diff
@@ -16,6 +16,7 @@ import { getWorkdaysTableColumns } from "@/components/elements/Inputs/DataTable/
 import { TreeSpeciesValue } from "@/components/elements/Inputs/TreeSpeciesInput/TreeSpeciesInput";
 import Text from "@/components/elements/Text/Text";
 import { FormSummaryProps } from "@/components/extensive/WizardForm/FormSummary";
+import { useVerifyDate } from "@/hooks/useVerifyDate";
```
Comment (pachonjcl) [inline src/components/extensive/WizardForm/FormSummaryRow.tsx:19]: remove and do not use hooks

File: src/admin/apiProvider/utils/dateFormat.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { format, isValid, parseISO } from "date-fns";
+
+export const convertDateformat = (value: any) => {
```
Comment (pachonjcl) [inline src/admin/apiProvider/utils/dateFormat.ts:3]: change the name of this function `convertDateFormat`

**Review by roguenet — APPROVED** [review]

**Pull request merged**